hack/generator/pkg/astmodel/code_generation_context.go

```diff
@@ -14,16 +16,22 @@ import "github.com/pkg/errors"
 type CodeGenerationContext struct {
 	packageImports map[PackageReference]PackageImport
 	currentPackage PackageReference
+
+	generatedPackages map[PackageReference]*PackageDefinition
 }
 
 // New CodeGenerationContext creates a new immutable code generation context
-func NewCodeGenerationContext(currentPackage PackageReference, packageImports map[PackageImport]struct{}) *CodeGenerationContext {
+func NewCodeGenerationContext(
+	currentPackage PackageReference,
+	packageImports map[PackageImport]struct{},
+	generatedPackages map[PackageReference]*PackageDefinition) *CodeGenerationContext {
+
 	packageImportsMap := make(map[PackageReference]PackageImport)
 	for imp := range packageImports {
 		packageImportsMap[imp.PackageReference] = imp
 	}
 
-	return &CodeGenerationContext{currentPackage: currentPackage, packageImports: packageImportsMap}
+	return &CodeGenerationContext{currentPackage: currentPackage, packageImports: packageImportsMap, generatedPackages: generatedPackages}
```

Comment:
I think this has reached the point where formatting across multiple lines might be doing.

Reply:
Agreed